pep-0646.rst

```diff
@@ -736,35 +736,45 @@ shape properties of numerical computing programs.
 Grammar Changes
 ===============
 
-This PEP would require two grammar changes. Full diffs of ``python.gram``
+This PEP requires two grammar changes. Full diffs of ``python.gram``
 and simple tests to confirm correct behaviour are available at
 https://github.com/mrahtz/cpython/commits/pep646-grammar.
 
 Star expressions in indexes
 ---------------------------
 
 The first grammar change enables use of star expressions in index operations (that is,
-within square brackets), necessary to support star-unpacking of TypeVarTuples.
+within square brackets), necessary to support star-unpacking of TypeVarTuples:
+
+::
+
+    DType = TypeVar('DType')
+    Shape = TypeVarTuple('Shape')
+    class Array(Generic[DType, *Shape]):
+        ...
+
+(Note that ``from future import annotations`` does not help here, since no annotations
+are involved.)
```

Comment:
This remark is a bit mysterious. Why would a reader make the mistake of thinking that would help? What might they think it helps for?

Also, it's `__annotations__`.

Reply:
`from __future__ import annotations`

Reply:
I added this based on one of your earlier comments:

> > The first grammar change enables use of star expressions in index operations (that is,
> > within square brackets), necessary to support star-unpacking of TypeVarTuples.
> 
> I think this should summarize the reason for this change here. The first Summary example shows it well:
> 
> >    class Array(Generic[DType, *Shape]):
> 
> That this is the very first example shows how fundamental this is. (And note that "from future import annotations" doesn't help in this position, since it is not technically an annotation.)

I thought you were gesturing towards the fact that with future annotations, new syntax is unnecessary for being able to do whatever manner of exotic things in `x: Tensor[A, B, C, D]` because of annotations not being evaluated at runtime - so a naive reader might think the same was true for everything else suggested in the PEP.

Anyway, I also thought it was a bit unclear in the end, so I'm happy to remove it.